ChatCore/Utils/TaskManager.swift

```diff
@@ -11,6 +11,44 @@ import BackgroundTasks
 
 // MARK: Helper class to automatically manage closures by applying various attributes
 final class TaskManager {
+    class TaskCache {
```

Comment:
can we pls add some comment why we have this internal class and why its necessary to have queue.sync to read/write from dictionary? thx